quinn-proto/src/transport_parameters.rs

```diff
@@ -387,45 +431,53 @@ impl TransportParameters {
             let len = len as usize;
 
             match id {
-                0x00 => decode_cid(len, &mut params.original_dst_cid, r)?,
-                0x02 => {
+                id if TransportParameterId::OriginalDestinationConnectionId == id => {
```

Comment:
One more thing! Could we do a `TryFrom<u64> for `TransportParameterId` impl with `Err = ()` so that we can simplify these match arms?